backend/src/AppServer.hs

```diff
@@ -51,17 +51,18 @@ submitReportHandler report = case validateReport report of
   Failure errors -> throwError $ err422 {errBody = show errors}
   Success (RawGameReport {..}) -> runDb $ do
     -- TODO Logging in this stupid monad
-    -- TODO Reduce code duplication
     timestamp <- liftIO getCurrentTime
     year <- currentYear
+
     winnerId <- insertPlayerIfNotExists winner
     loserId <- insertPlayerIfNotExists loser
-    reportId <- insertGameReport $ toGameReport timestamp winnerId loserId report
+
     winnerStats <- getStats winnerId year
     loserStats <- getStats loserId year
 
-    let winnerSide = side
-    let loserSide = case winnerSide of Free -> Shadow; Shadow -> Free
+    reportId <- insertGameReport $ toGameReport timestamp winnerId loserId report
+
+    let (winnerSide, loserSide) = (side, other side)
```

Comment:
Pretty ✨

Reply:
In the WotR game implementation code, I actually have this side/other side logic elevated into the type level! :star_struck: